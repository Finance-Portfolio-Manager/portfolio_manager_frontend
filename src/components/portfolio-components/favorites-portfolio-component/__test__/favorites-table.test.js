import React, { Component } from 'react';
import ReactDOM from 'react-dom';
import FavoritesTable from './../favorites-table';

import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";

import renderer from "react-test-renderer";

import { Switch, Route, BrowserRouter as Router } from 'react-router-dom';

import { shallow, ShallowWrapper, mount } from "enzyme";

import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

const tableRows = [{symbol: "TSLA", currentPrice: 5}, {symbol: "GME", currentPrice: 100}]

const toggleFn = jest.fn();

configure({ adapter: new Adapter() });
const wrapper = shallow(<FavoritesTable tableRows={tableRows} toggleShowDetails={toggleFn}></FavoritesTable>);

test("matches snapshot", ()=>{
    const tree = renderer.create(<FavoritesTable tableRows={tableRows} toggleShowDetails={toggleFn}></FavoritesTable>).toJSON();
    expect(tree).toMatchSnapshot();
})