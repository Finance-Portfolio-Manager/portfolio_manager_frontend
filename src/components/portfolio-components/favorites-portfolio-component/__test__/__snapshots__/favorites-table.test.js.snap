// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<div
  className="wrapper fadeInDown folioContent"
  id="folioContent"
>
  <div
    className="table-responsive"
  >
    <div
      className="table-responsive"
    >
      <table
        className="table borderless table-hover table"
        id="stocks-view"
      >
        <thead>
          <tr>
            <th
              scope="col"
            >
              Stock
            </th>
            <th
              scope="col"
            >
              Current Price
            </th>
          </tr>
        </thead>
        <tbody
          id="stocks-body"
        >
          <tr>
            <td>
              TSLA
            </td>
            <td>
              5
            </td>
          </tr>
          <tr>
            <td>
              GME
            </td>
            <td>
              100
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <button
      className="btn btn-primary"
      disabled={false}
      onClick={[MockFunction]}
      type="button"
    >
      Go to card
    </button>
  </div>
</div>
`;
